src/main/java/org/reaktivity/nukleus/kafka/internal/stream/NetworkConnectionPool.java

```diff
@@ -1069,7 +1070,12 @@ void doRequestIfNeeded()
                     }
                     else
                     {
-                        doFetchRequest();
+                        if (!inFetch)
+                        {
+                            inFetch = true;
+                            doFetchRequest();
+                        }
+                        inFetch = false;
```

Comment:
This seems to be protecting against a second call to `doFetch` via `doRequestIfNeeded` when already processing `doFetch`. Is a boolean sufficient, or do we need a counter that calls `doFetchRequest` when incremented from `0` and also when decrementing to `0` ?

Reply:
doFetchRequest() is getting called only from doRequestIfNeeded(). It seems like boolean is good enough.
Previously, I was doing the same thing for the entire doRequestIfNeeded() via a flag. But that runs into issues like doListOffsetsRequest() is not getting called sometimes. So moved the condition specific to doFetchRequest().